### ISSUE TITLE
[unicorn/no-array-for-each] rule suggested for pIteration.forEach not array forEach

### ISSUE THREAD

**Issue by OmgImAlexis** [sindresorhus/eslint-plugin-unicorn#1508]
I've added this plugin and it's suggesting I replace pIteration and not just the built in array's forEach, is this the correct behavior?

```
   95:20  error  Use `for…of` instead of `Array#forEach(…)`           unicorn/no-array-for-each
```

```ts
await pIteration.forEach(plugins, async pluginName => {
    // My other code...
});
```

**Comment by fregante** [comment]
Context: https://www.npmjs.com/package/p-iteration

It seems that the module will iterate each promise as it resolves, not in order, which is peculiar (and sounds useful in some cases)

**Comment by fisker** [comment]
Do you think ignore `pIteration.forEach` is good enough for your case?

**Comment by zloirock** [comment]
That sounds like a case for `Promise.all`.
